### PR TITLE
Install `dwarf_convert.py` and `dat_convert.py`

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lfortran/lfortran/issues/3190

### CODE REVIEW

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -1580,9 +1580,9 @@ int link_executable(const std::vector<std::string> &infiles,
             compile_cmd += " -l" + runtime_lib + " -lm";
             run_cmd = "./" + outfile;
         }
-	if (verbose) {
+        if (verbose) {
             std::cout << compile_cmd << std::endl;
-	}
+        }
```
Comment (junghans) [inline src/bin/lfortran.cpp:1585]: Unrelated change?

File: src/bin/CMakeLists.txt [hunk 2]
```diff
@@ -41,6 +41,13 @@ if (WITH_STACKTRACE AND APPLE AND CMAKE_CXX_COMPILER_ID MATCHES Clang)
             POST_BUILD
             COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../libasr/dwarf_convert.py lfortran.dSYM/raw.txt lfortran.dSYM/lines.txt lfortran.dSYM/lines.dat
         )
+
+        # Install the dwarf_convert.py and dat_convert.py
+        install(
+            FILES ../libasr/dwarf_convert.py ../libasr/dat_convert.py
+            PERMISSIONS OWNER_EXECUTE OWNER_READ
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lfortran
+        )
```
Comment (junghans) [inline src/bin/CMakeLists.txt:50]: I am a little confused why you added this here and not in the `src/libasr` dir?

File: src/bin/lfortran.cpp (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/bin/lfortran.cpp:1585]: Yes, unrelated.

File: src/bin/CMakeLists.txt (hunk 2 above)
Comment (Shaikh-Ubaid) [reply src/bin/CMakeLists.txt:50]: Do you mean why we install them in `${CMAKE_INSTALL_INCLUDEDIR}/lfortran`? Or do you mean why this `install()` command is present in this `CMakeLists.txt`?

I am not exactly sure of the best location for this `install()` command. I added it here since this was the only place in all the `CMakeLists.txt` where `dwarf_convert.py` was already referenced. Since it was already referenced/used (on line 42), I thought to have the install at the same place on the next line.

If you think any other location could be better, please let me know and I will be happy to update.

Comment (junghans) [reply src/bin/CMakeLists.txt:50]: Sorry, let me clarify, I don't like that we have the `install()` for a file from another directory here, that is just confusing, I think it would be better to put the `install()` block in `src/libasr/CMakeLists.txt`.

Comment (certik) [reply src/bin/CMakeLists.txt:50]: Yes, let's move this to `src/libasr/CMakeLists.txt`.

Comment (Shaikh-Ubaid) [reply src/bin/CMakeLists.txt]: Updated team!

**Comment by Shaikh-Ubaid** [comment]
Once this is complete/merged, I think we might need to make another release.

File: src/libasr/CMakeLists.txt [hunk 3]
```diff
@@ -112,3 +112,10 @@ endif()
 if (WITH_LLVM)
     target_link_libraries(asr p::llvm)
 endif()
+
+# Install the dwarf_convert.py and dat_convert.py
+install(
+    FILES dwarf_convert.py dat_convert.py
+    PERMISSIONS OWNER_EXECUTE OWNER_READ
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lfortran
```
Comment (junghans) [inline src/libasr/CMakeLists.txt:120]: Is the include directory the right location? I think BINDIR might be better.

And is this needed on Linux as well or just osx?

Comment (Shaikh-Ubaid) [reply src/libasr/CMakeLists.txt:120]: Thanks @junghans.

> And is this needed on Linux as well or just osx?

`lfortran` references these `dwarf_convert.py` and `dat_convert.py` here https://github.com/lfortran/lfortran/blob/24c894c64936f18d6119ba8514ab2aad77b6ae99/src/bin/lfortran.cpp#L1603-L1605

The above code seems to execute on both linux and mac. Hence at the moment we install these scripts on both linux and mac.

>Is the include directory the right location?

https://github.com/lfortran/lfortran/blob/24c894c64936f18d6119ba8514ab2aad77b6ae99/src/bin/lfortran.cpp#L1602-L1603

`lfortran` searchers for these scripts in the `libasr` directory which is a step back from the `runtime_library_c_header_dir`. The installed version of `lfortran` has the `runtime_library_c_header_dir` in `${CMAKE_INSTALL_INCLUDEDIR}/lfortran/impure` as below.

https://github.com/lfortran/lfortran/blob/24c894c64936f18d6119ba8514ab2aad77b6ae99/src/runtime/legacy/CMakeLists.txt#L43-L44

Thus, a step back from it would be `${CMAKE_INSTALL_INCLUDEDIR}/lfortran` which is the current location where we place the `dwarf_convert.py` and `dat_convert.py`. Installing them in this directory enables us to reuse the existing code that searches in the `libasr` directory. If we update the location, then we might need to also update
`lfortran` to search in the updated location.

Please let me know what we should do.

Comment (junghans) [reply src/libasr/CMakeLists.txt:120]: I see, I understand the reason, but from the Linux FHS perspective it is still the wrong location.

Comment (certik) [reply src/libasr/CMakeLists.txt:120]: What is the right location -- `BINDIR`? Isn't that just `$PREFIX/bin`? If so, it seems it would pollute the global bin PATH for the user. These two scripts are internal for LFortran. It would be better inside some "lfortran" subdirectory, whether `share` or another place.

Comment (junghans) [reply src/libasr/CMakeLists.txt:120]: `share` would be ok I guess

Comment (Shaikh-Ubaid) [reply src/libasr/CMakeLists.txt:120]: It seems `dwarf_convert.py` and `dat_convert.py` are more of a temporary solutions. For the moment, shall we reuse the `lfortran` support for searching in the `libasr` directory and thus use the existing approach in this PR?

Comment (certik) [reply src/libasr/CMakeLists.txt:120]: @Shaikh-Ubaid can you please put it into `$PREFIX/share/lfortran`?

Comment (certik) [reply src/libasr/CMakeLists.txt:120]: I would do it the proper way, so that LFortran can be readily packaged in distributions.

Comment (Shaikh-Ubaid) [reply src/libasr/CMakeLists.txt]: Updated!

**Review by certik — APPROVED** [review]
I think this is fine. I do not like that we call `get_executable_path` all the time, it would be better to call it once and deduce everything from it. That can be implemented later.

**Comment by Shaikh-Ubaid** [comment]
@certik  This is merged. I think we need to make another release.

**Comment by certik** [comment]
Let's review and merge today's batch of PRs and then release.

**Comment by certik** [comment]
Released: https://github.com/lfortran/lfortran/releases/tag/v0.33.0
